src/main/java/com/nextcloud/android/sso/api/NextcloudAPI.java

```diff
@@ -119,9 +118,12 @@ public void stop() {
         try (Reader targetReader = new InputStreamReader(responseStream)) {
             if (targetEntity != Void.class) {
                 return gson.fromJson(targetReader, targetEntity);
+            } else {
```

Comment:
This is a safe `ClassCastException`, since we are casting the result... Correct me if I'm wrong, but i think you cast the `Object` somewhere to `Void`, because I definitely don't. I used to get, what I defined I want to get, so the casting is done SSO-Internally, see comment below...

Reply:
Ugh.. me being naive while thinking that I could just throw in some typescript-style code 🤷‍♂️ 😅  I think you might be right about the casting thing. However it would be interesting to see if it actually fails 🙈

Reply:
Hm yes, I get that point. I think I'm also good with the nothing option then. However I'm just thinking about other case. So if you specify `List` without a type, it'll be interpreted as `Void` as well, right? But what if I'm too lazy as a programmer to write the type but I still want to somehow receive my response object..? Maybe I'm overthinking it a little here..

Reply:
> So if you specify List without a type, it'll be interpreted as Void as well, right? But what if I'm too lazy as a programmer to write the type but I still want to somehow receive my response object..?

Nope, would be a `List<Object>` you could cast, if your're brave enough to guess the actual type. Would work in this case. The problem with your solution is only the void type, which is broken here. So we're back at the beginning of our problem :rofl:. I'll have another look at it, maybe this weekend and see what I can do about this.